crates/topos/tests/cert_delivery.rs

```diff
@@ -309,3 +313,209 @@ async fn cert_delivery() {
         panic!("Timeout waiting for command");
     }
 }
+
+// Picks a random peer and sends it a certificate. All other peers listen for broadcast certs.
+// Three possible outcomes:
+// 1. No errors, returns Ok
+// 2. There were errors, returns a list of all errors encountered
+// 3. timeout
+async fn check_certificate_delivery(
```

Comment:
I know that this test isn't new and just get moved, but I'm not sure what is supposed to be the difference between the goal of this test and the `cert_delivery` test above in this same file. cc @Freyskeyd

Reply:
`check_certificate_delivery` checks that all nodes has delivered a certificate. `cert_delivery` checks that every subnet  should receive certificates that has cross chain transaction targeting it

Reply:
This method is covering the delivery on only one target subnet and test the full delivery.
Maybe we can refactor some parts to avoid code duplication, but they don't test the same thing.
Also, we could rename this method to `assert_certificate_full_delivery`?